packages/shared/src/components/fields/Dropdown.tsx

```diff
@@ -123,6 +126,12 @@ export function Dropdown({
 
   const fullScreen = openFullScreen ?? isMobile;
 
+  useLayoutEffect(() => {
+    if (wasVisible === 'true' && !isVisible) {
+      triggerRef?.current?.focus?.();
+    }
+  }, [isVisible]);
```

Comment:
I think we can avoid the additional hooks but just adding the `triggerRef?.current?.focus?.();` at the `onHidden` callback of the `Menu` component.

Reply:
Hi @sshanzel, thanks for comment!  I've tried, but unfortunately `onHidden` is called on the very first render of the `Menu` component, resulting in an autofocus behavior on the last dropdown on the page :(

Reply:
Oh, then I think there is something wrong with the callback. Anyways, we will replace our context menu completely in the future. I think spending more time to check why that happens is probably unnecessary. Should be good to go here! 🥳